src/main/java/com/bobocode/svydovets/beans/definition/ConfigurationBeanDefinition.java

```diff
@@ -0,0 +1,31 @@
+package com.bobocode.svydovets.beans.definition;
+
+import java.lang.reflect.Method;
+
+/**
+ * Extends BeanDefinition and contain information that will be further used to create beans instances that
+ * were declared in @Configuration classes.
+ */
+public class ConfigurationBeanDefinition extends BeanDefinition {
+
+    private Method beanMethod;
+
+    public ConfigurationBeanDefinition(String name, Class<?> beanClass, Method beanMethod) {
```

Comment:
I think these methods in the `BeanDefinition` or we would need to cast before using it.

Reply:
From one side - yes. Bean from the Configuration class will be the extension of BeanDefinition. 
But from the other side, in case we put Method beanMethod into BeanDefinition class, many beans will be annotated with Component annotations with NULL Method value.
And also in case, there's a need somehow modify or extend the functionality of Beans created in Configuration classes, it can be already done in ConfigurationBeanDefinition class not touching the original BeanDefinition class.

I'd prefer to discuss this moment before making any changes

Reply:
Spring has inside of [BeanDefinition](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/factory/config/BeanDefinition.html#getFactoryMethodName--).
I propose to have in one class for now if we see that we need to split we can do it later.